jenkins.groovy

```diff
@@ -6,25 +6,166 @@ pipeline
          }
       }
-
       stage('Linter and Integration Test')
       {
+         when
+         {
+           not
+           {
+            buildingTag()
+           }
+         }
          steps
          {
             sh 'make test'
          }
       }
+      stage('Deploy image')
+      {
+        when
+        {
+          buildingTag()
+        }
+
+        steps
+        {
+          echo "TAG_NAME: ${TAG_NAME}"
+
+          script
+          {
+            if (TAG_NAME =~ /^(?:[4-9]|\\d{2,}).[0-9]+.[0-9]+-(?:[0-9]+|alpha|beta|rc).*/)
+            {
+              PARSED_VERSION = parse_version(TAG_NAME)
+              MAJOR = PARSED_VERSION[0]
+              MINOR = PARSED_VERSION[1]
+              PATCH = PARSED_VERSION[2]
+              echo "MAJOR: ${MAJOR}"
+              echo "MINOR: ${MINOR}"
+              echo "PATCH: ${PATCH}"
+
+              docker.withRegistry('', 'docker-hub')
+              {
+                sh 'docker run --rm --privileged multiarch/qemu-user-static --reset -p yes'
+                sh 'docker buildx create --name multiarch --driver docker-container --use | true'
+                sh 'docker buildx inspect --bootstrap'
+                sh """
+                    docker buildx build \
+                      --platform linux/arm64/v8 \
+                      --no-cache \
+                      --build-arg GRAYLOG_VERSION=\$(cat VERSION) \
+                      --build-arg BUILD_DATE=\$(date -u +\"%Y-%m-%dT%H:%M:%SZ\") \
+                      --tag graylog/graylog:${env.TAG_NAME}-arm64 \
+                      --tag graylog/graylog:${MAJOR}.${MINOR}.${PATCH}-arm64 \
+                      --tag graylog/graylog:${MAJOR}.${MINOR}-arm64 \
+                      --file docker/oss/Dockerfile \
+                      --push \
+                      .
+                """
+
+                sh """
+                  docker buildx build \
+                  --platform linux/arm64/v8 \
+                  --no-cache \
+                  --build-arg GRAYLOG_VERSION=\$(cat VERSION) \
+                  --build-arg BUILD_DATE=\$(date -u +\"%Y-%m-%dT%H:%M:%SZ\") \
+                  --tag graylog/graylog-enterprise:${env.TAG_NAME}-arm64 \
+                  --tag graylog/graylog-enterprise:${MAJOR}.${MINOR}.${PATCH}-arm64 \
+                  --tag graylog/graylog-enterprise:${MAJOR}.${MINOR}-arm64 \
+                  --file docker/enterprise/Dockerfile \
+                  --push \
+                  .
+                """
+              }
+            }
+
+            if (TAG_NAME =~ /forwarder-.*/)
+            {
+              PARSED_VERSION = parse_forwarder_version(TAG_NAME)
+              MAJOR = PARSED_VERSION[0]
+              MINOR = PARSED_VERSION[1]
+              PATCH = PARSED_VERSION[2]
+              echo "MAJOR: ${MAJOR}"
+              echo "MINOR: ${MINOR}"
+              echo "PATCH: ${PATCH}"
+
+              docker.withRegistry('', 'docker-hub')
+              {
+                sh """
```

Comment:
For the other build above we execute 3 more buildx related commands. Are they not needed here?

Reply:
Yeah, good point. Those other commands really only need to be run once on each server. I just left them in there for good measure, just in case something went wrong and they needed to be run again.

Reply:
Alright, I added the setup commands to the forwarder stage as well.